### ISSUE TITLE
limit the number of simultaneous servers

### ISSUE THREAD

**Issue by rokroskar** [SwissDataScienceCenter/renku-notebooks#46]
When a user requests a new server, check how many are already running and if that number is equal to `MAX_USER_SERVERS` then shut down the oldest one before starting the new one.

**Comment by lorenzo-cavazzi** [comment]
this should be possible strarting from version `1.0` by using `c.JupyterHub.named_server_limit_per_user` https://jupyterhub.readthedocs.io/en/stable/changelog.html#new-features

**Comment by rokroskar** [comment]
We probably want to resolve https://github.com/SwissDataScienceCenter/renku-notebooks/issues/222 and https://github.com/SwissDataScienceCenter/renku-ui/issues/686 before imposing these limits.

**Comment by olevski** [comment]
Should this be moved to Amalthea?

**Comment by rokroskar** [comment]
Not sure I think it should be the notebook service concern since it's also the one initiating new sessions. Why would it go to amalthea? It doesn't really know the concept of a user afaik.

**Comment by olevski** [comment]
Yeah that is a good point @rokroskar I did not read the description well enough. I thought this talked about limiting the total number of all servers on a cluster. Not the number per user.
